addons/account/models/account_payment_term.py

```diff
@@ -21,6 +22,22 @@ def _default_line_ids(self):
     company_id = fields.Many2one('res.company', string='Company')
     sequence = fields.Integer(required=True, default=10)
 
+    # -------Early payment discount fields-------
+    has_early_payment = fields.Boolean(string=_("Apply Early Payment Discount"), default=False)
+    percentage_to_discount = fields.Float(string=_("Discount"), default=1)
+    discount_computation = fields.Selection([
+        ('included', _('Tax included')),
+        ('excluded', _('Tax excluded')),
+       ], string='Computation', default='included')
+    discount_days = fields.Integer(string=_('Availability'), required=True, default=1)
+    discount_time_availability = fields.Selection([
+        ('d_day_after_invoice_date', _("days after the invoice date")),
+        ('d_after_invoice_month', _("days after the end of the invoice month")),
+        ('d_day_following_month', _("of the following month")),
+        ('d_day_current_month', _("of the current month")),
+    ], default='d_day_after_invoice_date', required=True, string=_('Discount Availability'))
```

Comment:
```suggestion
    discount_time_availability = fields.Selection(
        string="Discount Availability",
        selection=[
            ('days_after_invoice_date', "days after the invoice date"),
            ('days_after_invoice_month', "days after the end of the invoice month"),
            ('day_following_month', "of the following month"),
            ('day_current_month', "of the current month"),
        ],
        default='days_after_invoice_date',
        required=True,
    )
```
note: your `d_` prefix looks weird :p , I edited my suggestion